### PR TITLE
Enable ZMQ between GNMI and Orchanget

### PR DESCRIPTION
**Pull request by liuh-80**
Enable ZMQ on gnmi and orchagent

#### Why I did it
Improve GNMI API performance for Dash resources

#### How I did it
Modify gnmi and orchagent service start script, add ZMQ parameter.

#### How to verify it
Pass all UT & E2E test
Manually verify with create Dash resources via gnmi API.

### CODE REVIEW

**Comment by liuh-80** [comment]
Depends PR merged, close and re-open to trigger rebuild.

**Review by ganglyu — APPROVED** [review]

File: dockers/docker-orchagent/orchagent.sh [hunk 1]
```diff
@@ -69,4 +69,6 @@ else
     ORCHAGENT_ARGS+="-m $MAC_ADDRESS"
 fi
 
+ORCHAGENT_ARGS+=" -q tcp://127.0.0.1:8100"
```
Comment (qiluo-msft) [inline dockers/docker-orchagent/orchagent.sh:72]: >ORCHAGENT_ARGS [](http://example.com/codeflow?start=0&length=14)

On non-dash related platforms, this new arg is not necessary. Could you limit the scope?

File: dockers/docker-sonic-telemetry/telemetry.sh [hunk 2]
```diff
@@ -70,6 +70,8 @@ else
     TELEMETRY_ARGS+=" -v=2"
 fi
 
+TELEMETRY_ARGS+=" -zmq_address=tcp://127.0.0.1:8100"
```
Comment (qiluo-msft) [inline dockers/docker-sonic-telemetry/telemetry.sh:73]: >TELEMETRY_ARGS [](http://example.com/codeflow?start=0&length=14)

Could you limit the scope?

File: dockers/docker-orchagent/orchagent.sh (hunk 1 above)
Comment (liuh-80) [reply dockers/docker-orchagent/orchagent.sh:72]: As I know there is no flag for Dash, I will check and update this PR.

**Review by qiluo-msft — APPROVED** [review]

File: dockers/docker-orchagent/orchagent.sh [hunk 3]
```diff
@@ -69,4 +69,10 @@ else
     ORCHAGENT_ARGS+="-m $MAC_ADDRESS"
 fi
 
+# Enable ZMQ for SmartSwitch
+LOCALHOST_SUBTYPE=`sonic-db-cli CONFIG_DB hget localhost "subtype"`
```
Comment (Pterosaur) [inline dockers/docker-orchagent/orchagent.sh:73]: Here should be
```
sonic-db-cli CONFIG_DB hget "DEVICE_METADATA|localhost" "subtype"
```

File: dockers/docker-sonic-telemetry/telemetry.sh [hunk 4]
```diff
@@ -70,6 +70,12 @@ else
     TELEMETRY_ARGS+=" -v=2"
 fi
 
+# Enable ZMQ for SmartSwitch
+LOCALHOST_SUBTYPE=`sonic-db-cli CONFIG_DB hget localhost "subtype"`
```
Comment (Pterosaur) [inline dockers/docker-sonic-telemetry/telemetry.sh:74]: Same as above

File: dockers/docker-orchagent/orchagent.sh (hunk 3 above)
Comment (liuh-80) [reply dockers/docker-orchagent/orchagent.sh:73]: Thanks, I will update with new PR

File: dockers/docker-sonic-telemetry/telemetry.sh (hunk 4 above)
Comment (liuh-80) [reply dockers/docker-sonic-telemetry/telemetry.sh:74]: Thanks, I will update with new PR

**Pull request merged**